system-addon/lib/SnippetsFeed.jsm

```diff
@@ -213,6 +228,12 @@ this.SnippetsFeed = class SnippetsFeed {
       case at.TOTAL_BOOKMARKS_REQUEST:
         this.getTotalBookmarksCount(action._target.browser);
         break;
+      case at.NEW_TAB_INIT:
+        this._onSessionStart();
+        break;
+      case at.NEW_TAB_UNLOAD:
```

Comment:
I'm wondering if maybe we should be look at a more general *browser* session, like maybe the `UNINIT` event, since `NEW_TAB_UNLOAD` happens frequently within the same browser session

Reply:
oh yeah! that's great. should we keep the currentSessionStart or dump it? or set it to `INIT` instead of `NEW_TAB_INIT`?